### PR TITLE
Use DOCKER_CONFIG environment variable to look up auth config

### PR DESCRIPTION
**Pull request by shin-**
When provided, default paths (`~/.docker/config.json` and `~/.dockercfg`) are ignored.

### CODE REVIEW

File: docker/auth/auth.py [hunk 1]
```diff
@@ -136,10 +136,19 @@ def load_config(config_path=None):
...
 
+    if not config_path and os.environ.get('DOCKER_CONFIG'):
+        config_path = os.path.join(
+            os.environ.get('DOCKER_CONFIG'),
+            os.path.basename(DOCKER_CONFIG_FILENAME)
+        )
+
     # Prefer ~/.docker/config.json.
     config_file = config_path or os.path.join(os.path.expanduser('~'),
                                               DOCKER_CONFIG_FILENAME)
```
Comment (dnephin) [inline docker/auth/auth.py]: I'd like to make a suggestion here that I think will make it easier to test.  I think the logic for "find config file" can be separated from "read contents of config file".

```python

def find_config(config_path=None):
    for path in (
        config_path, 
        os.path.join(os.environ.get('DOCKER_CONFIG'), os.path.basename(DOCKER_CONFIG_FILENAME)),
        os.path.join(os.path.expanduser('~'), DOCKER_CONFIG_FILENAME),
        os.path.exists(os.path.join(os.path.expanduser('~'), LEGACY_DOCKER_CONFIG_FILENAME))
    ):
        if path and os.path.exists(path):
            return path

    return None
```

File: tests/test.py [hunk 2]
```diff
@@ -2423,6 +2423,37 @@ def test_load_config_with_random_name(self):
...
+                'email': 'sakuya@scarlet.net'
+            }
+        }
+
+        with open(dockercfg_path, 'w') as f:
+            json.dump(config, f)
+
+        environ_mem = os.environ.copy()
+        try:
+            os.environ['DOCKER_CONFIG'] = folder
```
Comment (dnephin) [inline tests/test.py]: This could be done with 

```python

with mock.patch.dict(os.environ):
    ...

```

**Comment by shin-** [comment]
Implemented @dnephin 's suggestions and rebased.

**Comment by dnephin** [comment]
LGTM

**Pull request merged**